plugins/inputs/disk/disk.go

```diff
@@ -53,13 +54,24 @@ func (ds *DiskStats) Gather(acc telegraf.Accumulator) error {
 			// Skip dummy filesystem (procfs, cgroupfs, ...)
 			continue
 		}
+
+		device := partitions[i].Device
 		mountOpts := MountOptions(partitions[i].Opts)
 		tags := map[string]string{
 			"path":   du.Path,
-			"device": strings.ReplaceAll(partitions[i].Device, "/dev/", ""),
+			"device": strings.ReplaceAll(device, "/dev/", ""),
 			"fstype": du.Fstype,
 			"mode":   mountOpts.Mode(),
 		}
+
+		m := strings.Count(device, "/dev/")
+		if m > 0 {
+			label, _ := disk.Label(strings.Replace(device, "/dev/", "", 1))
+			if label != "" {
+				tags["label"] = label
+			}
+		}
```

Comment:
Wouldn't this be more clear
```suggestion
		label, err := disk.Label(strings.TrimPrefix(device, "/dev/"))
		if err == nil && label != "" {
			tags["label"] = label
		}
```

Reply:
agreed, committed